packages/back-end/src/services/experiments.ts

```diff
@@ -397,17 +401,31 @@ export function determineNextDate(schedule: ExperimentUpdateSchedule | null) {
   return new Date(Date.now() + hours * 60 * 60 * 1000);
 }
 
-export async function createSnapshot(
-  experiment: ExperimentInterface,
-  user: EventAuditUser,
-  phaseIndex: number,
-  organization: OrganizationInterface,
-  dimensionId: string | null,
-  useCache: boolean = false,
-  statsEngine: StatsEngine | undefined,
-  regressionAdjustmentEnabled: boolean,
-  metricRegressionAdjustmentStatuses: MetricRegressionAdjustmentStatus[]
-) {
+export async function createSnapshot({
+  experiment,
+  organization,
+  user = null,
+  phaseIndex,
+  dimension = null,
+  useCache = false,
+  statsEngine,
+  regressionAdjustmentEnabled,
+  metricRegressionAdjustmentStatuses,
+  sequentialTestingEnabled,
+  sequentialTestingTuningParameter,
+}: {
+  experiment: ExperimentInterface;
+  organization: OrganizationInterface;
+  user?: EventAuditUser;
+  phaseIndex: number;
+  dimension?: string | null;
+  useCache?: boolean;
+  statsEngine?: StatsEngine;
+  regressionAdjustmentEnabled?: boolean;
+  metricRegressionAdjustmentStatuses?: MetricRegressionAdjustmentStatus[];
+  sequentialTestingEnabled?: boolean;
+  sequentialTestingTuningParameter?: number;
```

Comment:
Two general question heres (with this and RA)...

We used to pass so much of the config via the ExperimentInterface (things like the attribution model, or the phase data), but now we're adding separate arguments for config, these four:
```
  regressionAdjustmentEnabled?: boolean;
  metricRegressionAdjustmentStatuses?: MetricRegressionAdjustmentStatus[];
  sequentialTestingEnabled?: boolean;
  sequentialTestingTuningParameter?: number;
  ```
  
  1. Is the reason we have them separate largely because they can be set *outside* the context of an experiment (e.g. the org settings?) or is there some other compelling reason to keep them out of the experiment interface? Given that we're already saying that org settings are defaults *for new experiments that are created*, is there a way to just do this in the Experiment Interface? Maybe not, just was wondering and wanted to keep that conversation open.
  2. Is it worth it to refactor these to an `ExperimentConfig` or something like that if they (as I likely assume) need to remain separate? I think it's now a total of 5 arguments and I wonder if a separate type is worth investing in.

Reply:
Did some more digging. I think pushing these settings through the experimentInterface is not currently safe. A side effect of calling `createSnapshot()` is we do a comparison of the current experimentInterface (including the fields/overrides we pushed in) versus the previous one and then updating the SDK payload and logging the before/after changes. We'd end up corrupting those.

But a config object could work.

Reply:
> we do a comparison of the current experimentInterface

Yeah, that makes sense. Thanks for looking in to it.

> But a config object could work.

Yeah, just an idea! But maybe better handled by project scoped settings.